src/fsharp/block.fs

```diff
@@ -0,0 +1,222 @@
+module Internal.Utilities.Library.Block
+
+open System.Collections.Immutable
+
+type block<'T> = ImmutableArray<'T>
+type blockbuilder<'T> = ImmutableArray<'T>.Builder
+
+[<RequireQualifiedAccess>]
+module BlockBuilder =
+
+    let create size : blockbuilder<'T> =
+        if size = 0 then
+            ImmutableArray.CreateBuilder()
+        else
+            ImmutableArray.CreateBuilder(size)
```

Comment:
As it is this code means that size 0 returns a builder with default capacity (8), I assume that's what is intended?

Reply:
Probably not, if I set it to 0, the builder should be 0.